system/zpool.py

```diff
@@ -0,0 +1,374 @@
+  bootfs:
+    description:
+      - Identifies the default bootable dataset for the root pool.
+    required: False
+  cachefile:
+    description:
+      - Sets the path to where the pool configuration is cached
+    required: False
+  comment:
+    description:
+      - The comment property
+    required: False
+  dedupditto:
+    description:
+      - Sets a threshold for number of copies. If the reference count for a deduplicated block goes above this threshold, another ditto copy of the block is stored automatically
+    required: False
+  failmode:
+    description:
+      - The failmode property
+    required: False
+    choices: ['wait','continue','panic']
+  feature@async_destroy:
+    description:
+      - Enable or disable asynchronous filesystem and volume destruction. Features cannot be disabled once they have been enabled
+    required: False
+    choices: ['enabled]
+  feature@empty_bpobj:
+    description:
+      - Enable or disable the empty_bpobj feature. Features cannot be disabled once they have been enabled
+    required: False
+    choices: ['enabled]
+  feature@lz4_compress:
+    description:
+      - Enable or disable the lz4_compress feature. Features cannot be disabled once they have been enabled
+    required: False
+    choices: ['enabled]
+  force:
+    description:
+      - Supply -f flag, forcing creation of the zpool
+    type: Bool
+  l2arc_dev:
+    description:
+      - The device to use as the l2arc
+    required: False
+  listsnapshots:
+    description:
+      - The listsnapshots property
+    required: False
+    choices: ['on', 'off']
+  raid_type:
+    description:
+      - ZFS RAID type to use for zpool
+    choices: ['mirror','raidz','raidz2','raidz3']
+  spare_devs:
+    description:
+      - The spare device definition
+    required: False
+  version:
+    description:
+      - Specify zpool version
+    required: False
+  zil_dev:
+    description:
+      - The device to use for the ZFS intent log
+    required: False
+  zpool_devs:
+    description:
+      - The devices to use for the zpool
+    required: False
+author: Stan Hiller, based on ZFS module from Johan Wiren
+'''
+
+EXAMPLES = '''
+# Create a new RAIDZ zpool called rpool using sdc, sdd, and sdc
+- zpool: name=rpool state=present raid_type=raidz zpool_devs='sdc sdd sde'
+
+# Create a new RAIDZ2 zpool called rpool using sdc, sdd, and sdc, ashift=12, l2arc on sda7, and zil on sdb7
+- zpool: name=rpool raid_type=raidz2 state=present zpool_devs='sdc sdd sde' l2arc_dev=sda7 zil_dev=sdb7 ashift=12
+
+# Create a new RAIDZ zpool called rpool with a mirrored ZIL
+- zpool: name=rpool raid_type=raidz state=present zpool_devs='sdc sdd sde' zil_dev='mirror sdb7 sda7'
+
+# Destroy a zpool named rpool
+- zpool: name=rpool state=absent
+
+# Add sdb to rpool
+- zpool: name=rpool zpool_devs='sdb' state=add
+
+# Create a new RAID10 zpool called rpool using sda, sdb, sdc and sdd
+- zpool: name=rpool state=present raid_type=mirror zpool_devs='sda sdb'
```

Comment:
I would recommend removing all the *_devs properties and combine them into a general vdevs_specs property where you simply put in zpool arguments. that way you could also use it to setup a pool consisting of two mirrors.